app/models/response_node.rb

```diff
@@ -4,15 +4,35 @@
 class ResponseNode < ApplicationRecord
   self.table_name = "answers"
 
+  attr_accessor :relevant
+
   belongs_to :form_item, inverse_of: :answers, foreign_key: "questioning_id"
   belongs_to :response
   has_closure_tree order: "new_rank", numeric_order: true, dont_order_roots: true, dependent: :destroy
 
   before_save do
-    children.each { |c| c.response_id = response_id }
+    children.each do |c|
+      if !c.relevant? || c._destroy
+        c.destroy
+      else
+        c.response_id = response_id
+      end
+    end
+    # update inst_nums. inst_num and this block will go away with answer_arranger
+    children.reject(&:destroyed?).sort_by(&:new_rank).each_with_index do |c, i|
```

Comment:
Then let's put this in a third method and mark it with a TODO so we can easily search and destroy it.